### PR TITLE
Add missing course information to application show view

### PR DESCRIPTION
**Pull request by willmcb**
## Context

Adds more course information (accredited body and funding type) to the course table in the applications show view.

## Changes proposed in this pull request

**Before:**
Screenshot 2020-05-05 at 14 33 29

**After:**
Screenshot 2020-05-05 at 14 32 29

## Guidance to review

- If a course has no `accredited_provider` it is assumed that they are self accrediting and the provider name is rendered as the Accrediting bod (see spec); is this a correct assumption to make?
- Are the details in the right order?
- Is funding type the right thing to show to the user?
- are the names for the funding types correct? Underlying data for funding types is `apprenticeship`, `fee`, and `salary` which are rendered to the user as 'Apprenticeship', 'Fee paying' and 'Salaried' respectively.

## Link to Trello card

https://trello.com/c/Nj7rrX7e/2059-single-application-view-include-missing-information-in-course-section

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: spec/components/provider_interface/course_details_component_spec.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  it 'renders the study mode' do
+    expect(render.css('.govuk-summary-list__row')[5].text).to include('Study mode')
+    expect(render.css('.govuk-summary-list__row')[5].text).to include(application_choice.course_option.study_mode.humanize)
+  end
+
+  it 'renders financing funding type of a course' do
+    expect(render.css('.govuk-summary-list__row')[6].text).to include('Funding type')
+    expect(render.css('.govuk-summary-list__row')[6].text).to include('Fee paying')
+  end
```
Comment (tijmenb) [inline spec/components/provider_interface/course_details_component_spec.rb:59]: I appreciate the thoroughness of this testing!

Two questions:

- When we make changes in the ordering, we would need to change a lot of the assertions. Can we avoid that somehow?
- Some of the assertions duplicate the code that's being tested. For example, the study mode test in essence tests that `application_choice.course_option.study_mode.humanize` is the same as `application_choice.course_option.study_mode.humanize`. Perhaps you could make it specific, so we test that the resulting study mode string actually makes sense (and isn't a sneaky nil for example).

File: spec/factories.rb [hunk 2]
```diff
@@ -275,6 +275,10 @@
     trait :part_time do
       study_mode { :part_time }
     end
+
+    trait :accredited_provider do
+      accredited_provider { provider }
+    end
```
Comment (tijmenb) [inline spec/factories.rb:281]: Optional: this could go into the `let` where we're using it, to avoid the factories from becoming too large.

**Comment by adamsilver** [comment]
Please change ‘Study mode’ to ‘Full or part time’

File: spec/factories.rb (hunk 2 above)
Comment (willmcb) [reply spec/factories.rb]: done 👍

**Comment by willmcb** [comment]
> Please change ‘Study mode’ to ‘Full or part time’

File: spec/components/provider_interface/course_details_component_spec.rb (hunk 1 above)
Comment (willmcb) [reply spec/components/provider_interface/course_details_component_spec.rb]: have added a method that looks rows up by name and returns an index to make it easier to change row orders in future (they are now all in one place).

Comment (willmcb) [reply spec/components/provider_interface/course_details_component_spec.rb]: I have added literals to the test now 👍

Comment (willmcb) [reply spec/components/provider_interface/course_details_component_spec.rb]: Also moved to instance_doubles as there is no need for DB creation with this kind of component test in my mind.

**Pull request merged**